packages/agreement-process/src/services/agreementService.ts

```diff
@@ -997,27 +1004,34 @@ export function agreementServiceBuilder(
           suspendedByPlatform,
         });
 
-      const updatedAgreement: Agreement = {
+      const updatedAgreementWithoutContract: Agreement = {
         ...agreement.data,
         ...updatedAgreementSeed,
       };
 
+      const updatedAgreement: Agreement = await addContractOnFirstActivation(
+        isFirstActivation,
```

Comment:
minor personal opinion: it's a but odd that a function receives a flag stating if the function itself will be executed or not.
Probably when refactoring and splitting the two activation functions it will automatically become cleaner